recipes/up-cpp/all/conanfile.py

```diff
@@ -0,0 +1,159 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd, stdcpp_library
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file, rm, rmdir, collect_libs
+from conan.tools.microsoft import check_min_vs, is_msvc, is_msvc_static_runtime
+from conan.tools.scm import Version, Git
+import os
+
+required_conan_version = ">=1.59.0"
+
+class UpCppConan(ConanFile):
+    name = "up-cpp"
+    description = "C++ language implementation of a uProtocol language SDK"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/eclipse-uprotocol/up-cpp"
+    topics = ("uprotocol", "sdv", "middleware")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    # TODO: Upon release of up-spec 1.5.8, this will need to be updated to:
+    # proto_containing_folder = "up-spec"
+    proto_containing_folder = "up-core-api"
+    # TODO: Upon release of up-spec 1.5.8, this will need to be updated to:
+    # proto_subfolder = "up-core-api"
+    proto_subfolder = "."
+    proto_folder = "uprotocol"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_testing": [True, False],
+        "build_unbundled": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": False,
+        "build_testing": False,
+        "build_unbundled": False,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 14
+
+    # unclear on what I should choose for msvc and Visual Studio
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": "13",
+            "clang": "13",
+            "gcc": "11",
+            "msvc": "191",
+            "Visual Studio": "15",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+        self.folders.build = os.path.join("build", str(self.settings.build_type))
+        self.folders.install = "install"
+
+    def build_requirements(self):
+        self.build_requires("cmake/[>=3.18.0]")
+
+    def requirements(self):
+        self.requires("protobuf/3.21.12")
+        self.requires("spdlog/1.13.0")
+        if self.options.build_testing:
+            self.requires("gtest/1.14.0")
```

Comment:
```suggestion
        if self.options.build_testing:
            self.requires("gtest/1.14.0")
```

Testing is not something really supported, because the idea is packaging the project, not acting like a build tool. Still, in case you really want to be able to run unit tests, please, use the follow approach: https://github.com/conan-io/conan-center-index/blob/master/recipes/tree-gen/all/conanfile.py#L33 It will not affect the package ID, which is really important.